tools/cldr-apps/src/test/java/org/unicode/cldr/web/api/TestLocaleCompletion.java

```diff
@@ -0,0 +1,225 @@
+        }
+
+        final int LOCALE_SIZE = 1000; // We expect at least this many items.
+
+        // ** Step one.  Setup the locale
+        // Expected 100%
+        localeFile.add("//ldml/characters/exemplarCharacters", "[abc]");
+        localeFile.add("//ldml/numbers/symbols[@numberSystem=\"latn\"]/group",
+            ","); // ok
+        localeFile.add("//ldml/localeDisplayNames/languages/language[@type=\"" + TEST_LOCALE + "\"]",
+            "baccacab"); // for fallback (need to use only 'a,b,c')
+
+        localeFile.add("//ldml/numbers/currencies/currency[@type=\"XTS\"]/displayName[@draft=\"provisional\"]",
+            "Test Currency (comprehensive level, should not have an effect on completion)");
+        localeFile.add("//ldml/numbers/currencies/currency[@type=\"XUA\"]/displayName[@draft=\"unconfirmed\"]",
+            "Other Test Currency (comprehensive level, should not have an effect on completion)");
+
+        tc.invalidateAllCached();
+
+        {
+            LocaleCompletionResponse completion = LocaleCompletion.getLocaleCompletion(locale, stf);
+
+            assertNotNull(completion);
+            assertAll("tests on completio - 100%",
+                () -> assertEquals(Level.MODERN.name(), completion.level, "completion level"),
+                () -> assertTrue(completion.votes > LOCALE_SIZE, "completion votes should be bigger but was " + completion.votes),
+                () -> assertEquals(completion.total, completion.votes,
+                    "completion votes should be equal completion total but was  " + completion.votes + "/" + completion.total),
+                () -> assertTrue(completion.total > LOCALE_SIZE, "completion total should be bigger but was " + completion.total),
+                () -> assertEquals(0, completion.error),
+                () -> assertEquals(0, completion.missing),
+                () -> assertEquals(0, completion.provisional));
+        }
+
+        // ** Step two.
+        // Make some changes, and we should have a couple of errors.
+        // "Expected <100%"
+        // Mutate
+        localeFile.add("//ldml/characters/exemplarCharacters", "[]");
+        localeFile.add("//ldml/numbers/symbols[@numberSystem=\"latn\"]/group",
+            ",,,"); // err; expected 1, not 3
+        // rootFile.add("//ldml/localeDisplayNames/languages/language[@type=\""+TEST_LOCALE+"\"]",
+        //     TEST_LOCALE); // for fallback
+        localeFile.remove("//ldml/localeDisplayNames/languages/language[@type=\"" + TEST_LOCALE + "\"]");
+        localeFile.add("//ldml/dates/timeZoneNames/zone[@type=\"Pacific/Kanton\"]/exemplarCity[@draft=\"provisional\"]", "cabaca");
+        localeFile.add("//ldml/dates/timeZoneNames/zone[@type=\"Asia/Qyzylorda\"]/exemplarCity[@draft=\"unconfirmed\"]", "bacaba");
+        tc.invalidateAllCached();
+        final int EXPECT_ERROR = 2;
+        final int EXPECT_MISSING = 1;
+        final int EXPECT_PROVISIONAL = 2;
```

Comment:
The above results in 2 errors, 1 missing, 2 provisional.

The front end doesn't need the M/E/P split out, but it's useful for debugging.